cmd/kube-controller-manager/app/discovery.go

```diff
@@ -23,22 +23,15 @@ package app
 import (
 	"net/http"
 
-	discoveryv1alpha1 "k8s.io/api/discovery/v1alpha1"
-	"k8s.io/klog"
 	endpointslicecontroller "k8s.io/kubernetes/pkg/controller/endpointslice"
 )
 
 func startEndpointSliceController(ctx ControllerContext) (http.Handler, bool, error) {
-	if !ctx.AvailableResources[discoveryv1alpha1.SchemeGroupVersion.WithResource("endpointslices")] {
```

Comment:
still need to check if the beta API is available... beta APIs can be turned off and components should still run

Reply:
and check if the EndpointSlice feature is enabled before running